django/cantusdb_project/main_app/views/views.py

```diff
@@ -10,6 +11,7 @@
 )
 from main_app.forms import ContactForm
 from django.core.mail import send_mail, get_connection
+from requests import request
```

Comment:
I don't think this import is needed. The `request` referenced in `request.build_absolute_uri()` is the HttpRequest object in Django, instead of an external function. See documentation at https://docs.djangoproject.com/en/4.0/ref/request-response/#django.http.HttpRequest.build_absolute_uri